### PR TITLE
enableDefenderForPostgresql

### PR DESCRIPTION
**Pull request by fatima99s**

### CODE REVIEW

File: plugins/azure/defender/enableDefenderForPostgresql.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+                    'Unable to query for Pricing: ' + helpers.addError(pricings), location);
+                return rcb();
+            }
+
+            if (!pricings.data.length) {
+                helpers.addResult(results, 0, 'No Pricing information found', location);
+                return rcb();
+            }
+
+            helpers.checkMicrosoftDefender(pricings, 'opensourcerelationaldatabases', 'PostgreSQL Flexible Servers', results, location);
```
Comment (mehakseedat63) [inline plugins/azure/defender/enableDefenderForPostgresql.js:36]:
```suggestion
            helpers.checkMicrosoftDefender(pricings, 'opensourcerelationaldatabases', 'PostgreSQL Flexible Servers', results, location);
```
this plan is for multiple types of databases, we should remove PostgreSQL Flexible Server from the title and make this plugin generic for open source relational databases

**Review by mehakseedat63 — CHANGES_REQUESTED** [review]

File: exports.js [hunk 2]
```diff
@@ -995,10 +995,11 @@ module.exports = {
         'enableDefenderForStorage'      : require(__dirname + '/plugins/azure/defender/enableDefenderForStorage.js'),
         'enableDefenderForContainers'   : require(__dirname + '/plugins/azure/defender/enableDefenderForContainers.js'),
         'enableDefenderForSqlServers'   : require(__dirname + '/plugins/azure/defender/enableDefenderForSqlServers.js'),
+        'enableDefenderForOSRD'         : require(__dirname + '/plugins/azure/defender/enableDefenderForOSRD.js'),
         'enableEndpointIntegration'     : require(__dirname + '/plugins/azure/defender/enableEndpointIntegration.js'),
         'enableDefenderForDNS'          : require(__dirname + '/plugins/azure/defender/enableDefenderForDNS.js'),
         'enableDefenderForKeyVaults'     : require(__dirname + '/plugins/azure/defender/enableDefenderForKeyVaults.js'),
-
+
```
Comment (alphadev4) [inline exports.js:1002]:
```suggestion

```

File: plugins/azure/defender/enableDefenderForOSRD.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
+var async = require('async');
+var helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'Enable Defender For Open Source Relational Databases',
+    category: 'Defender',
+    domain: 'Management and Governance',
+    description: 'Ensures that Microsoft Defender is enabled for Open Source Relational Databases.',
+    more_info: 'Enabling Defender for Cloud on Open Source Relational Databases allows detection of unusual database access, query patterns, and suspicious activities, enhancing overall security.',
```
Comment (alphadev4) [inline plugins/azure/defender/enableDefenderForOSRD.js:9]:
```suggestion
    more_info: 'Enabling Defender for Cloud on Open Source Relational Databases allows detection of unusual database access, query patterns, and suspicious activities, enhancing overall security. This plan brings threat protections for PostgreSQL, MySQL and MariaDB  Azure Databases.'
```

**Review by alphadev4 — APPROVED** [review]

**Review by mehakseedat63 — APPROVED** [review]

File: helpers/azure/functions.js [hunk 4]
```diff
@@ -357,7 +357,7 @@ function checkServerConfigs(servers, cache, source, location, results, serverTyp
 }
 
 function checkMicrosoftDefender(pricings, serviceName, serviceDisplayName, results, location ) {
-   
+
```
Comment (alphadev4) [inline helpers/azure/functions.js:360]:
```suggestion
   
```

**Pull request merged**